File: ExamenMoria-Python/projecto/models/salasPython/SalaMagica.py
from Sala import Sala

class SalaMagica(Sala):
    def __init__(self, id: int, poderMaligno: int):
        super().__init__(id)
        self.poderMaligno = poderMaligno

    def __str__(self) -> str:
        return f"SalaMágica({self.numeroSala}, {self.poderMaligno})"

File: ExamenMoria_Java/src/main/java/models/salas/SalaMagica.java
package models.salas;

public class SalaMagica extends Sala{

    public int poderMaligno;

    public SalaMagica(int numeroSala, int poderMaligno) {
        super(numeroSala);
        this.poderMaligno = poderMaligno;
    }

    @Override
    public String toString() {
        return "SalaMagica{" +
                "poderMaligno=" + poderMaligno +
                ", numeroSala=" + numeroSala +
                '}';
    }
}
